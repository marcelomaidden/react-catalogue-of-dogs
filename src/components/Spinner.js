import React from 'react';

const Spinner = () => (
  <div className="spinner-border m-auto" role="status">
    <span className="sr-only" />
  </div>
);

export default Spinner;
