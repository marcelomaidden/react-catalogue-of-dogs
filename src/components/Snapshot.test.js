import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";

import Spinner from "./Spinner";

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it("should render a Spinner component", () => {
  act(() => {
    render(
      <Spinner />, 
      container
    );
  });
  expect(container.innerHTML).toMatchSnapshot(); 
});
