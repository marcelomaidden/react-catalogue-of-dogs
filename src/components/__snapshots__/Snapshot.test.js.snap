// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a Spinner component 1`] = `"<div class=\\"spinner-border m-auto\\" role=\\"status\\"><span class=\\"sr-only\\"></span></div>"`;
